pkg/kritis/signer/signer.go

```diff
@@ -78,23 +82,36 @@ var (
 // SignImage signs an image without doing any policy check.
 // Returns an error if creating an attestation fails.
 func (s Signer) SignImage(image string) error {
-	existed, _ := s.isAttestationAlreadyExist(image)
+	existed, err := s.isAttestationAlreadyExist(image)
+	if err != nil {
+		return fmt.Errorf("checking existing attestation status failed: %v", err)
+	}
 	if existed {
```

Comment:
Optional: I suggest restructuring the control flow to reduce nesting, bail out early, and make it easier to follow, maybe like this:
if exists && !overwrite {
  return ...
}
if exists && overwrite {
  delete ...
} else {
  doesn't exist
}
create attestation

Reply:
Done.